### PR TITLE
Rianfowler/add next-app-version hook and integrate into DashboardVersionCard

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- removes `app` and `isHelmManaged` props from `DashboardVersionCard` and replaces them with the `useSelectedApp` and `useIsHelmManaged` hooks
- removes `/next-app-version` call from `DashboardVersionCard` and replaces with `useNextAppVersionWithIntercept` hook

#### Which issue(s) this PR fixes:
[sc-50084]
https://app.shortcut.com/replicated/story/50084/fix-multi-app-ux

#### Does this PR introduce a user-facing change?
```release-note
- improves multiapp experience by caching and refetching next-app-version for faster re-renders of next app version card on dashboard 
```

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/features/Dashboard/components/DashboardVersionCard.tsx [hunk 1]
```diff
@@ -42,15 +44,13 @@ type Props = {
...
-  app: App;
   checkingForUpdates: boolean;
   checkingForUpdateError: boolean;
   checkingUpdateText: string;
   currentVersion: Version | null;
   downloadCallback: () => void;
   downstream: Downstream | null;
   isBundleUploading: boolean;
-  isHelmManaged: boolean;
   links?: string[];
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/DashboardVersionCard.tsx:54]: removed these props and added their respective hooks

File: web/src/features/Dashboard/components/DashboardVersionCard.tsx [hunk 2]
```diff
@@ -87,7 +87,6 @@ type State = {
   kotsUpdateMessage: string | null;
   kotsUpdateRunning: boolean;
   kotsUpdateStatus: VersionStatus | null;
-  latestDeployableVersion: Version | null;
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/DashboardVersionCard.tsx:90]: removed and added new custom hook for this

File: web/src/features/Dashboard/components/DashboardVersionCard.tsx [hunk 3]
```diff
@@ -168,6 +166,15 @@ const DashboardVersionCard = (props: Props) => {
...
   const params = useParams<KotsParams>();
+  const { selectedApp } = useSelectedApp();
+  const {
+    data: newAppVersionWithInterceptData,
+    error: latestDeployableVersionErrMsg,
+  } = useNextAppVersionWithIntercept(selectedApp?.slug || "");
+  const { latestDeployableVersion } = newAppVersionWithInterceptData || {};
+
+  const { data: isHelmManagedResponse } = useIsHelmManaged();
+  const { isHelmManaged = false } = isHelmManagedResponse || {};
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/DashboardVersionCard.tsx:177]: this is getting a little messy. I'm still into the idea of returning the `react-query` api as is just so there's a standard interface but may need to come up with a cleaner way to query some of our nested data responses. maybe we can un-nest them with `react-query`

File: web/src/features/Dashboard/components/DashboardVersionCard.tsx [hunk 4]
```diff
@@ -202,12 +209,23 @@ const DashboardVersionCard = (props: Props) => {
...
+      setState({
+        latestDeployableVersionErrMsg: `Failed to get latest deployable version: ${latestDeployableVersionErrMsg.message}`,
+      });
+      return;
+    }
+
+    setState({
+      latestDeployableVersionErrMsg: "Something went wrong, please try again.",
+    });
+  }, [latestDeployableVersionErrMsg]);
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/DashboardVersionCard.tsx:223]: This is just pulling the error handling that was in the fetch handler that we replaced with the ook

File: web/src/features/Dashboard/api/useNextAppVersion.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { useRef, useState, useEffect } from "react";
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/useNextAppVersion.ts:1]: This file probably looks familiar since I just copy pasta'd it. We can refine this later.

**Review by rianfowler — COMMENTED** [review]
important changes noted. The rest of this PR is updating file locations and import referneces.

**Review by miaawong — APPROVED** [review]
amazing 🙌

**Review by alicenstar — APPROVED** [review]

**Pull request merged**